packages/vue-app/template/client.js

```diff
@@ -491,6 +491,7 @@ function fixPrepatch(to, ___) {
       if (
         instance.constructor._dataRefresh &&
         Components[i] === instance.constructor &&
+        instance.$vnode.data.keepAlive !== true &&
```

Comment:
Just to be sure, is the default value of `keepAlive`, `false` by default? If it is true, isn't this a minor breaking change?

Reply:
The default value is `undefined`, it's just a flag in Vue for `keepAlive` cached component.

I don't think it's a breaking change, it should be a bug because `component` under `keep-alive` should be cached and never reload data as Vue is doing.

Reply:
Just wondering, shouldn't it be `data.props.keepAlive` instead of `data.keepAlive`? Because in bunch of files (`packages/vue-app/template/components/nuxt.js`, etc), `keepAlive` property is being wrapped into `props` object.

I ran into an issue when component is still trying to update its data() even though it has `keep-alived` enabled.
Changing this line to `data.props.keepAlive` solved problem for me.

PS: Nuxt v2.15.2 (seems like nothing have been changed with keep-alived behaviour ever since this commit)

Reply:
keepAlive here is checking vue keep-alive component which will add keepAlive to vnode.data https://github.com/vuejs/vue/blob/b51430f598b354ed60851bb62885539bd25de3d8/src/core/components/keep-alive.js#L120

Reply:
Thanks for your reply.

Sorry, but what should be the keep-alive component in here? Because debugger says that `instance` is a page component. Also, `instance.$vnode.data` doesn't have `keepAlive` property, but it does have `props.keepAlive`.

Screenshot:

![20210301_015410](https://user-images.githubusercontent.com/4021140/109426469-281e6980-7a31-11eb-93f5-bd154a14bbc4.png)

Reply:
I think the component in your screenshot is NuxtChild which has keep-alive prop for enabling keepAlive on it.

https://nuxtjs.org/docs/2.x/features/nuxt-components#keep-alive

Reply:
You right, it is NuxtChild component.
I thought it should work the same way for any components, but seems like it is not the case.
I'll try to investigate it a little bit further and maybe come back with new issue and/or PR.
Thanks for your comments 👍